src/vm/ceeload.h

```diff
@@ -2594,14 +2594,12 @@ class Module
     LPCWSTR GetDebugName() { WRAPPER_NO_CONTRACT; return m_file->GetDebugName(); }
 #endif
 
-    BOOL IsILOnly() { WRAPPER_NO_CONTRACT; return m_file->IsILOnly(); }
-
 #ifdef FEATURE_PREJIT
     BOOL HasNativeImage() 
     { 
         WRAPPER_NO_CONTRACT;
         SUPPORTS_DAC;
-        return m_file->HasNativeImage(); 
+        return m_file->HasNativeImage() && !IsCollectible();
```

Comment:
Why do you need this check? I think the check in `ReadyToRunInfo::Initialize` should be sufficient.

Reply:
I have thought that the ReadyToRunInfo::Initialize is used just for R2R. But we need to disable non-R2R crossgened binaries too, don't we?